pkg/ipam/allocator/podcidr/podcidr.go

```diff
@@ -849,12 +851,33 @@ func (n *NodesPodCIDRManager) allocateNext(nodeName string) (*nodeCIDRs, bool, e
 		nCIDRs.v6PodCIDRs = []*net.IPNet{v6CIDR}
 	}
 
+	if nCIDRs == nil {
+		err = fmt.Errorf("%s%s Unable allocate node cidr for node %s, Please check your configuration is correct",
+			getCIDRAllocatorsInfo(n.v4CIDRAllocators, v4AllocatorType),
+			getCIDRAllocatorsInfo(n.v4CIDRAllocators, v4AllocatorType),
```

Comment:
It should be n.v4CIDRAllocators and n.v6CIDRAllocators, It can show both ipv4 CIDR info and  IPV6 CIDR info if it's not nil.
The out put look like this
```
IPAMInfo(IPv4):
clusterCIDR: 10.0.0.0/24, nodeMask: 24
 Unable to allocate node CIDR for node node-1, please check that your configuration is correctIPAMInfo(IPv4):
```

Reply:
This will end up printed to the logs, I'd rather have it on a single line to make it easier to extract the whole line at once by grepping and all.
```
Unable to allocate node CIDR for node node-1. IPv4: [IPAMInfo: ..., clusterCIDR: 10.0.0.0/24, nodeMask: 24]. Please check...
```
The rest looks good, thanks for the changes!

Reply:
How about like this. We should consider also possibe have another protocol like IPv6. I think this is much better.
```
Unable to allocate node CIDR for node node-1. IPAMInfo: {IPv4: [clusterCIDR: 10.0.0.0/24, nodeMask: 24, clusterCIDR: 10.0.1.0/24, nodeMask: 24], IPv6: []}. Please check that your configuration is correct
```

Reply:
Sure, this one looks great! :+1:

Reply:
done